src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/EnumGroupAndName.cs

```diff
@@ -28,7 +30,23 @@ public EnumGroupAndName(string group, string name)
             }
 
             Group = group;
-            Name = name;
+            _nameFunc = () => name;
+        }
+
+        public EnumGroupAndName(string group, Func<string> name)
```

Comment:
I couldn't decide if we should `Obsolete` the old constructor. My instinct was no, to allow people to do the simple thing if they didn't need anything complex, but I fear that will need to ignoring the new constructor and getting us right back into the same spot.

Also, going back to add comments to this an any new public methods now.